addons/mail/models/mail_shortcode.py

```diff
@@ -20,3 +20,47 @@ class MailShortcode(models.Model):
         help="Content that will automatically replace the shortcut of your choosing. This content can still be adapted before sending your message.")
     description = fields.Char('Description')
     last_used = fields.Datetime('Last Used', help="Last time this shortcode was used")
+    group_ids = fields.Many2many("res.groups", string="Authorized Groups", groups="base.group_system")
+    user_ids = fields.Many2many("res.users", string="Authorized Users", groups="base.group_system")
+    is_shared = fields.Boolean(string="Determines if the shortcode is currently shared with other users", compute="_compute_is_shared", store=True)
+    is_editable = fields.Boolean(string="Determines if the shortcode can be edited by the current user", compute="_compute_is_editable")
+
+    @api.depends('create_uid')
+    def _compute_is_editable(self):
+        for shortcode in self:
+            if self.env.is_admin() or shortcode.create_uid.id == self.env.uid or not shortcode.id:
+                shortcode.is_editable = True
+            else:
+                shortcode.is_editable = False
+
+    @api.depends("group_ids", "user_ids")
+    def _compute_is_shared(self):
+        for shortcode in self:
+            shortcode.is_shared = True if shortcode.group_ids or shortcode.user_ids else False
+
+    def create(self, vals):
+        res = super().create(vals)
+        payload = {"CannedResponse": res._read_format(["id", "source", "substitution"])}
+        notifications = [(self.env.user.partner_id, "mail.record/insert", payload)]
+        for group in res.group_ids:
```

Comment:
are you sending a notification on all groups for all shortcodes ? Groups are not necessarily the same on all shortcodes created.